### PR TITLE
feat(installer): add thanos support to tke-installer

### LINKED ISSUE #1100: tke-installer支持安装thanos组件
**What would you like to be added**:

tke-installer在监控设置支持安装thanos组件选项
image

image
https://github.com/tkestack/tke/blob/master/docs/guide/zh-CN/installation/installation-architecture.md#monitor--notify-%E7%BB%84%E4%BB%B6

**Why is this needed**:

### PR DESCRIPTION
**Pull request by t1mt**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #1100

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by t1mt** [comment]
POST http://<tke-installer-host>/api/cluster
```json
{
    "cluster": {
        "apiVersion": "platform.tkestack.io/v1",
        "kind": "Cluster",
        "spec": {
            "networkDevice": "eth1",
            "features": {
                "enableMetricsServer": true
            },
            "dockerExtraArgs": {},
            "kubeletExtraArgs": {},
            "apiServerExtraArgs": {},
            "controllerManagerExtraArgs": {},
            "schedulerExtraArgs": {},
            "clusterCIDR": "192.168.0.0/16",
            "properties": {
                "maxClusterServiceNum": 256,
                "maxNodePodNum": 256
            },
            "type": "Baremetal",
            "machines": [
                {
                    "ip": "<YOUR_MACHINE_IP>",
                    "port": <SSH_PORT>,
                    "username": "<LOGIN_USER>",
                    "password": "<PASSWORD | base64>"
                }
            ]
        }
    },
    "config": {
        "basic": {
            "username": "<TKE_LOGIN_NAME>",
            "password": "<TKE_LOGIN_PASSWORD | base64>"
        },
        "auth": {
            "tke": {}
        },
        "registry": {
            "tke": {
                "domain": "registry.tke.com"
            }
        },
        "application": {},
        "business": {},
        "monitor": {
            "thanos": {
                "bucketConfig": {
                    "type": "s3",
                    "config": {
                        "access_key": "<ACCESS_KEY>",
                        "bucket": "<BUCKET_NAME>",
                        "endpoint": "<IP:PORT>",
                        "insecure": true,
                        "secret_key": "<SECRET_KEY>",
                        "signature_version2": true
                    }
                }
            }
        },
        "logagent": {},
        "gateway": {
            "domain": "console.tke.tiger.com",
            "cert": {
                "selfSigned": {}
            }
        }
    }
}
```

File: cmd/tke-installer/app/installer/images/images.go [hunk 1]
```diff
@@ -71,6 +72,7 @@ var components = Components{
 	Registry: containerregistry.Image{Name: "registry", Tag: "2.7.1"},
 	Busybox:  containerregistry.Image{Name: "busybox", Tag: "1.31.1"},
 	InfluxDB: containerregistry.Image{Name: "influxdb", Tag: "1.7.9"},
+	Thanos:   containerregistry.Image{Name: "thanos", Tag: "v0.15.0"},
```
Comment (huxiaoliang) [inline cmd/tke-installer/app/installer/images/images.go:75]: image should per-push to docker hub

File: cmd/tke-installer/app/installer/manifests/thanos/thanos.yaml [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+    - name: http
+      port: 10902
+      protocol: TCP
+      targetPort: 10902
+      nodePort: 30902
+    - name: remote-write
+      port: 19291
+      protocol: TCP
+      targetPort: 19291
+      nodePort: 30291
```
Comment (huxiaoliang) [inline cmd/tke-installer/app/installer/manifests/thanos/thanos.yaml:64]: hard code nodeport will cause port conflict

**Review by kevinfeng — APPROVED** [review]
/LGTM

File: cmd/tke-installer/app/installer/manifests/thanos/thanos.yaml [hunk 3]
```diff
@@ -0,0 +1,584 @@
...
+              name: thanos-receive-hashrings
+              subPath: thanos-receive-hashrings.json
+      terminationGracePeriodSeconds: 120
+      volumes:
+        - configMap:
+            defaultMode: 420
+            name: thanos-receive-hashrings
+          name: thanos-receive-hashrings
+        - name: data
+          hostPath:
```
[the remark this answers is not in the document]
Comment (t1mt) [reply cmd/tke-installer/app/installer/manifests/thanos/thanos.yaml:180]: StatefulSet

**Pull request merged**